src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMSource.java

```diff
@@ -440,13 +440,15 @@ protected final void retrieve(@CheckForNull SCMSourceCriteria criteria,
             String fullName = repoOwner + "/" + repository;
             final GHRepository repo = github.getRepository(fullName);
             listener.getLogger().format("Looking up %s%n", HyperlinkNote.encodeTo(repo.getHtmlUrl().toString(), fullName));
-            try {
-                repositoryUrl = repo.getHtmlUrl();
-                collaboratorNames = new HashSet<>(repo.getCollaboratorNames());
-            } catch (FileNotFoundException e) {
-                // not permitted
-                listener.getLogger().println("Not permitted to query list of collaborators, assuming none");
-                collaboratorNames = Collections.emptySet();
+            if (!repo.isPrivate()) { // random people cannot create pull requests to private repositories
```

Comment:
The idea is that if you have write access to the repository then your PRs can be trusted.

People can have read access and submit PRs.

Consider the case where OPS exposes their configuration as code.

They accept PRs from anyone, but they want to control the PRs getting merged.

Somebody could submit a Jenkinsfile that when run with the production secrets could have catastrophic effects (and they may not realise... need not be malicious)

So OPS allow people to read (to provide better understanding) and to create PRs (to reduce the work OPS have to do) but they want to review changes before being built.

I am not saying that the above is the majority case, but we cannot just break that usage pattern without fair warning as people who have relied on that behaviour would be understandably quite upset.